Remotion/Data/DomainObjects/Infrastructure/DomainObjectTransactionContextIndexer.cs

```diff
@@ -25,26 +26,20 @@ namespace Remotion.Data.DomainObjects.Infrastructure
   /// </summary>
   public struct DomainObjectTransactionContextIndexer
   {
-    private readonly DomainObject _domainObject;
-    private readonly bool _isInitializedEventExecuting;
+    private readonly DomainObjectTransactionContextImplementation _implementation;
 
-    public DomainObjectTransactionContextIndexer (DomainObject domainObject, bool isInitializedEventExecuting)
+    public DomainObjectTransactionContextIndexer (DomainObjectTransactionContextImplementation implementation)
     {
-      ArgumentUtility.CheckNotNull("domainObject", domainObject);
-      _domainObject = domainObject;
-      _isInitializedEventExecuting = isInitializedEventExecuting;
+      ArgumentUtility.CheckNotNull("domainObject", implementation.DomainObject);
```

Comment:
arg check should be for impllementation, not implementation.domainobject. it may be okay to also make that check debug-only.